src/pages/api/aws/sign-s3.ts

```diff
@@ -0,0 +1,49 @@
+import {NextApiRequest, NextApiResponse} from 'next'
+import AWS from 'aws-sdk'
+import {v4 as uuidv4} from 'uuid'
+
+const options = {
+  bucket: process.env.AWS_VIDEO_UPLOAD_BUCKET,
+  region: process.env.AWS_VIDEO_UPLOAD_REGION,
+  signatureVersion: 'v4',
+  ACL: 'public-read',
+}
+
+const credentials = {
+  accessKeyId: process.env.AWS_VIDEO_UPLOAD_ACCESS_KEY_ID || '',
+  secretAccessKey: process.env.AWS_VIDEO_UPLOAD_SECRET_ACCESS_KEY || '',
+}
+
+AWS.config.credentials = credentials
+
+const s3 = new AWS.S3(options)
+
+const signedUrl = async (req: NextApiRequest, res: NextApiResponse) => {
+  if (req.method === 'GET') {
```

Comment:
Any permissions checking we want to do here? Otherwise this endpoint is wide open to anyone who wants to upload to the S3 bucket.

Reply:
probably, I don't know exactly how that'd be accomplished so I just left it. feels like we are inching towards the "big sack of booleans" for permissions, but outside of giving people ideas here in this channel, I doubt the risk is significant because there is literally nothing to gain from shoving stuff into this bucket that I can tell.

https://www.npmjs.com/package/accesscontrol

https://dev.to/noclat/unify-acl-across-front-end-and-back-end-with-next-js-and-nextauth-js-5ban

This looks way better than the "is role" style to me and something we should probably consider.